pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -92,7 +93,12 @@ func (p *Plugin) SetLogger(log hclog.Logger) {
 
 // Configure sets up the plugin
 func (p *Plugin) Configure(ctx context.Context, req *plugin.ConfigureRequest) (*plugin.ConfigureResponse, error) {
-	config, err := p.validateConfig(req.Configuration)
+	// cancels previous schedule delete task in case of re configure
+	if p.cancelScheduleDelete != nil {
+		p.cancelScheduleDelete()
+	}
```

Comment:
We should only cancel this after we're sure the configure call is going to succeed and we're setting up the next deleter.